chalk-rust-ir/src/lib.rs

```diff
@@ -130,8 +143,15 @@ pub struct TraitDatumBound<I: Interner> {
 pub struct TraitFlags {
     pub auto: bool,
     pub marker: bool,
+    /// Indicate that a trait is defined upstream (in a dependency), used during
+    /// coherence checking.
     pub upstream: bool,
+    /// The trait equivalent of a struct fundamental, currently (2020-03-25)
```

Comment:
A fundamental trait is a trait where adding an impl for an existing type is considered a breaking change. Examples of fundamental traits are the closure traits like `Fn` and `FnMut`. As of this writing, fundamental traits are declared by the unstable `#[fundamental]` attribute in rustc, and hence cannot appear outside of the standard library.